lib/src/code_controller.dart

```diff
@@ -62,13 +92,10 @@ class CodeController extends TextEditingController {
     ],
     this.webSpaceFix = true,
     this.onChange,
-  }) : super(text: text) {
-    // PatternMap
-    if (language != null && theme == null) throw Exception("A theme must be provided for language parsing");
```

Comment:
If we allow to change language and theme, then I suggest we allow language without theme. If they both present, then we will format the code. If either one is not present, then not.

Otherwise there will be too much checks, and a nasty method like `setThemeAndLanguage` to set both in one call.